### ISSUE TITLE
False positibe Lint/Syntax on pattern matching and splat operator applied to method result

### ISSUE THREAD

**Issue by flash-gordon** [rubocop/rubocop#14056]
This code

```ruby
x => ^([*a.x])
```

triggers a syntax error `Lint/Syntax: a.x' is not allowed as a local variable name`

--------

## Expected behavior

It's allowed syntax, no error is expected.

## Actual behavior

Rubocop gets confused about this.

## Steps to reproduce the problem

```
$ sed -i '' 's/2\.7/3\.4/g' .rubocop.yml
$ echo "x => ^([*a.x])" > example.rb
$ rubocop example.rb

Inspecting 1 file
F

Offenses:

example.rb:1:10: F: Lint/Syntax: a.x' is not allowed as a local variable name
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops`)
x => ^([*a.x])
         ^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
1.75.1 (using Parser 3.3.7.3, Prism 1.4.0, rubocop-ast 1.43.0, analyzing as Ruby 3.4, running on ruby 3.4.2) [arm64-darwin24]
  - rubocop-performance 1.24.0
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.5.0
```

**Comment by Earlopain** [comment]
Thanks for the report, I openend https://github.com/ruby/prism/pull/3529 which should fix this

**Comment by Earlopain** [comment]
Will be fixed with the next prism release (1.4.1 or 1.5.0)

**Comment by Earlopain** [comment]
prism 1.5.0 got released. It seems to work ok from my tests and accepts the code now, so this can be closed.
